### PR TITLE
[stable/datadog] Add PSP and SCC

### PR DESCRIPTION
**Pull request by clamoriniere**
#### What this PR does / why we need it:

Add the possibility to create a `PodSecurityPolicy` and a `SecurityContextConstraints` resource for the Agent's pods.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/datadog/README.md [hunk 1]
```diff
@@ -370,7 +370,14 @@ helm install --name <RELEASE_NAME> \
...
+| `agents.podSecurity.capabilites`                             | list of allow capabilities                                                                                                                                                          | `[SYS_ADMIN, SYS_RESOURCE, SYS_ADMIN, IPC_LOCK]`|
+| `agents.podSecurity.volumes`                                 | list of allow volumes types                                                                                                                                                         | `[configMap,downwardAPI,emptyDir,ostPath,secret]`|
+| `agents.podSecurity.seccompProfiles`                         | List of allow seccomp profiles                                                                                                                                                      | `["*"]`                                         |
+| `agents.podSecurity.apparmorProfiles`                        | List of allow apparmor profiles                                                                                                                                                     | `["*"]`                                         |
```
Comment (L3n41c) [inline stable/datadog/README.md]:
```suggestion
| `agents.podSecurity.capabilites`                             | list of allowed capabilities                                                                                                                                                        | `[SYS_ADMIN, SYS_RESOURCE, SYS_ADMIN, IPC_LOCK]`|
| `agents.podSecurity.volumes`                                 | list of allowed volumes types                                                                                                                                                       | `[configMap,downwardAPI,emptyDir,ostPath,secret]`|
| `agents.podSecurity.seccompProfiles`                         | List of allowed seccomp profiles                                                                                                                                                    | `["*"]`                                         |
| `agents.podSecurity.apparmorProfiles`                        | List of allowed apparmor profiles                                                                                                                                                   | `["*"]`                                         |
```

File: stable/datadog/README.md [hunk 2]
```diff
@@ -370,7 +370,14 @@ helm install --name <RELEASE_NAME> \
...
+| `agents.podSecurity.podSecurityPolicy.create`                | If true, create a PodSecurityPolicy resource for the Agent's Pods                                                                                                                   | `False`                                         |
+| `agents.podSecurity.securityContextConstraints.create`       | If true, create a SecurityContextConstraints resource for the Agent's Pods                                                                                                          | `False`                                         |
+| `agents.podSecurity.privileged`                              | If true, allow privileged containers                                                                                                                                                | `False`                                         |
+| `agents.podSecurity.capabilites`                             | list of allow capabilities                                                                                                                                                          | `[SYS_ADMIN, SYS_RESOURCE, SYS_ADMIN, IPC_LOCK]`|
```
Comment (vboulineau) [inline stable/datadog/README.md]: allow -> allowed (also in values.yaml)

File: stable/datadog/templates/agent-psp.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  allowedCapabilities: 
+{{ toYaml .Values.agents.podSecurity.capabilites | indent 4 }}
+  volumes:
+{{ toYaml .Values.agents.podSecurity.volumes | indent 4 }}
+  fsGroup:
+    rule: RunAsAny
+  runAsUser:
+    rule: RunAsAny
+  seLinux:
+    rule: RunAsAny
```
Comment (L3n41c) [inline stable/datadog/templates/agent-psp.yaml]: Shouldn’t we have the same value as the one we have for OpenShift in the SCC ?
If we need a particular SElinux context on OpenShift, we should need the same for vanilla kubernetes on SElinux-enabled distribution.

```suggestion
  seLinux:
    rule: {{ .Values.datadog.securityContext.type }}
    seLinuxOptions:
      {{ toYaml .Values.datadog.securityContext.seLinuxOptions | indent 6 }}
```

File: stable/datadog/templates/agent-scc.yaml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+seccompProfiles:
+{{ toYaml .Values.agents.podSecurity.seccompProfiles | indent 2 }}
+allowedCapabilities:
+{{ toYaml .Values.agents.podSecurity.capabilites | indent 2 }}
+#
+# The rest is copied from restricted SCC
+#
+allowHostDirVolumePlugin: true
+allowHostIPC: false
+allowPrivilegedContainer: false
```
Comment (L3n41c) [inline stable/datadog/templates/agent-scc.yaml]: In the PSP, use of privileged containers is configurable.
Do we want have the same option for PSP and SCC ?

```suggestion
allowPrivilegedContainer: {{ .Values.agents.podSecurity.privileged }}
```

File: stable/datadog/templates/agent-scc.yaml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+seLinuxContext:
+{{- if .Values.datadog.securityContext }}
+{{ toYaml .Values.datadog.securityContext | indent 2 }}
+{{- else }}
+  type: MustRunAs
+  seLinuxOptions:
+    user: "system_u"
+    role: "system_r"
+    type: "spc_t"
+    level: "s0"
```
Comment (L3n41c) [inline stable/datadog/templates/agent-scc.yaml]: If we explicitly set the default value in `values.yaml`, it would allow us to remove that `if`.
```suggestion
{{ toYaml .Values.datadog.securityContext | indent 2 }}
```

File: stable/datadog/values.yaml [hunk 6]
```diff
@@ -48,6 +48,7 @@ datadog:
   ## modifying the label type below:
   #
   securityContext: {}
+  #   type: MustRunAs
   #   seLinuxOptions:
   #     user: system_u
   #     role: system_r
```
Comment (L3n41c) [inline stable/datadog/values.yaml]: If we explicitly set the default value here, it will simplify the templates by removing the need to test `nil` value and to put the default value in the template.

```suggestion
  securityContext:
    type: MustRunAs
    seLinuxOptions:
      user: system_u
      role: system_r
      type: spc_t
      level: s0
```

File: stable/datadog/values.yaml [hunk 7]
```diff
@@ -665,6 +666,65 @@ agents:
...
+      #
+      create: false
+
+    ## @param privileged - boolean - optional
+    ## If true, Allow to run privileged containers
+    #
+    privileged: false
+
+    ## @param capabilites - list - optional
+    ## Allow capabilites
```
Comment (L3n41c) [inline stable/datadog/values.yaml]:
```suggestion
    ## Allowed capabilites
```

File: stable/datadog/values.yaml [hunk 8]
```diff
@@ -665,6 +666,65 @@ agents:
...
+    #
+    capabilites:
+    - SYS_ADMIN
+    - SYS_RESOURCE
+    - SYS_PTRACE
+    - NET_ADMIN
+    - IPC_LOCK
+
+    ## @param volumes - list - optional
+    ## Allow volumes types
```
Comment (L3n41c) [inline stable/datadog/values.yaml]:
```suggestion
    ## Allowed volumes types
```

File: stable/datadog/values.yaml [hunk 9]
```diff
@@ -665,6 +666,65 @@ agents:
...
+    #
+    volumes:
+    - configMap
+    - downwardAPI
+    - emptyDir
+    - hostPath
+    - secret
+
+    ## @param seccompProfiles - list - optional
+    ## Allow seccomp profiles
```
Comment (L3n41c) [inline stable/datadog/values.yaml]:
```suggestion
    ## Allowed seccomp profiles
```

File: stable/datadog/values.yaml [hunk 10]
```diff
@@ -665,6 +666,65 @@ agents:
...
+    #
+    seccompProfiles:
+    - "*"
+
+    ## @param apparmorProfiles - list - optional
+    ## Allow apparmor profiles
```
Comment (L3n41c) [inline stable/datadog/values.yaml]:
```suggestion
    ## Allowed apparmor profiles
```

File: stable/datadog/values.yaml [hunk 11]
```diff
@@ -665,6 +666,65 @@ agents:
...
+    - downwardAPI
+    - emptyDir
+    - hostPath
+    - secret
+
+    ## @param seccompProfiles - list - optional
+    ## Allowed seccomp profiles
+    #
+    seccompProfiles:
+    - "*"
```
Comment (L3n41c) [inline stable/datadog/values.yaml]: Do we want do restrict the default list of allowed seccomp profiles ?

```suggestion
    - "runtime/default"
    - "localhost/system-probe"
```

We should also add to this list `{{ .Values.datadog.system-probe.seccomp }}`, but I’m not sure this is possible inside `values.yaml`.

**Review by L3n41c — APPROVED** [review]

File: stable/datadog/values.yaml (hunk 11 above)
Comment (clamoriniere) [reply stable/datadog/values.yaml]: good idea
